custom/shared/contexts/MediaDeviceProvider.js

```diff
@@ -1,82 +1,63 @@
-import React, { createContext, useContext, useCallback } from 'react';
+import React, { createContext, useContext, useMemo } from 'react';
+import { useDevices, useLocalParticipant } from '@daily-co/daily-react-hooks';
 import PropTypes from 'prop-types';
 
-import { useCallState } from './CallProvider';
-import { useParticipants } from './ParticipantsProvider';
-import { useDevices } from './useDevices';
+export const DEVICE_STATE_LOADING = 'loading';
+export const DEVICE_STATE_PENDING = 'pending';
+export const DEVICE_STATE_ERROR = 'error';
+export const DEVICE_STATE_GRANTED = 'granted';
+export const DEVICE_STATE_NOT_FOUND = 'not-found';
+export const DEVICE_STATE_NOT_SUPPORTED = 'not-supported';
+export const DEVICE_STATE_BLOCKED = 'blocked';
+export const DEVICE_STATE_IN_USE = 'in-use';
+export const DEVICE_STATE_OFF = 'off';
+export const DEVICE_STATE_PLAYABLE = 'playable';
+export const DEVICE_STATE_SENDABLE = 'sendable';
 
 export const MediaDeviceContext = createContext();
 
 export const MediaDeviceProvider = ({ children }) => {
-  const { callObject } = useCallState();
-  const { localParticipant } = useParticipants();
-
   const {
-    camError,
-    cams,
-    currentCam,
-    currentMic,
-    currentSpeaker,
-    deviceState,
-    micError,
-    mics,
-    refreshDevices,
-    setCurrentCam,
-    setCurrentMic,
-    setCurrentSpeaker,
+    hasCamError,
+    cameras,
+    camState,
+    setCamera,
+    hasMicError,
+    microphones,
+    micState,
+    setMicrophone,
     speakers,
-  } = useDevices(callObject);
+    setSpeaker,
+    refreshDevices,
+  } = useDevices();
 
-  const selectCamera = useCallback(
-    async (newCam) => {
-      if (!callObject || newCam.deviceId === currentCam?.deviceId) return;
-      const { camera } = await callObject.setInputDevicesAsync({
-        videoDeviceId: newCam.deviceId,
-      });
-      setCurrentCam(camera);
-    },
-    [callObject, currentCam, setCurrentCam]
-  );
+  const localParticipant = useLocalParticipant();
 
-  const selectMic = useCallback(
-    async (newMic) => {
-      if (!callObject || newMic.deviceId === currentMic?.deviceId) return;
-      const { mic } = await callObject.setInputDevicesAsync({
-        audioDeviceId: newMic.deviceId,
-      });
-      setCurrentMic(mic);
-    },
-    [callObject, currentMic, setCurrentMic]
-  );
+  const isCamMuted = useMemo(() => {
```

Comment:
How come we are re-defining these here when we also have the same code in `BasicTray.js`? Is there any opportunity to reuse these, or do you think the duplication is a bit better for readability for the purpose of these examples?